src/common/meta/src/key/datanode_table.rs

```diff
@@ -197,18 +208,19 @@ impl DatanodeTableManager {
                 opts.push(TxnOp::Delete(raw_key))
             }
         }
-
+        let need_update_options = current_region_options != new_region_options;
```

Comment:
`current_region_options` is useless, regardless of whether `current` is equal to `new`, it is actually using the `new`.